src/MissionManager/MissionController.cc

```diff
@@ -754,20 +806,54 @@ void MissionController::_recalcAltitudeRangeBearing()
             if (!item->isStandaloneCoordinate()) {
                 firstCoordinateItem = false;
                 if (lastCoordinateItem != homeItem || (showHomePosition && linkBackToHome)) {
-                    double azimuth, distance, altDifference;
+                    double azimuth, distance, altDifference, telemetryDistance;
 
                     // Subsequent coordinate items link to last coordinate item. If the last coordinate item
                     // is an invalid home position we skip the line
                     _calcPrevWaypointValues(homePositionAltitude, item, lastCoordinateItem, &azimuth, &distance, &altDifference);
                     item->setAltDifference(altDifference);
                     item->setAzimuth(azimuth);
                     item->setDistance(distance);
+
+                    missionDistance += distance;
+
+                    if (item->isSimpleItem()) {
+                        _calcHomeDist(item, homeItem, &telemetryDistance);
+
+                        if (vtolCalc) {
+                            SimpleMissionItem* simpleItem = qobject_cast<SimpleMissionItem*>(item);
+                            if (simpleItem->command() == MavlinkQmlSingleton::MAV_CMD_NAV_TAKEOFF || hoverDistanceCalc){
+                                hoverDistance += distance;
+                            }
+                            cruiseDistance = missionDistance - hoverDistance;
+                            if(simpleItem->command() == MavlinkQmlSingleton::MAV_CMD_NAV_LAND && !linkBackToHome && !cruiseTransition && !hoverTransition){
+                                hoverDistance = cruiseDistance;
+                                cruiseDistance = missionDistance - hoverDistance;
+                            }
+                        }
+                    } else {
+                        missionDistance += qobject_cast<ComplexMissionItem*>(item)->surveyDistance();
```

Comment:
@AndKe That's true! Thank you so much for pointing this out, hadn't considered this before. I will prepare a PR today to tackle this.